modules/samples/producers/wso2-event-performance/src/main/java/org/wso2/carbon/sample/performance/Client.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.sample.performance;
+
+import org.apache.log4j.Logger;
+import org.wso2.carbon.databridge.agent.AgentHolder;
+import org.wso2.carbon.databridge.agent.DataPublisher;
+import org.wso2.carbon.databridge.commons.Event;
+
+import java.text.DecimalFormat;
+import java.util.Arrays;
+import java.util.Random;
+
+public class Client {
+    private static Logger log = Logger.getLogger(Client.class);
+
```

Comment:
1. Actually for this case we are not doing a calculation of all the events. We initially specify event gaps (like 1000 events) and calculate performance between 1000 intervals and log the through put for that each interval. In the end there will be multiple logged calculations. In that case do you think we need a warming up?

2. Since we are doing calculation by comparing event sending and receiving separately,  yes we can add nanoTime as you suggested if it increases the accuracy?